examples/website/plot/plot-layer/surface-layer.ts

```diff
@@ -109,20 +126,20 @@ export default class SurfaceLayer extends Layer {
    *   0--------> 1
    *              x
    */
-  encodePickingColor(i) {
+  encodePickingColor(i: number) {
     const {uCount, vCount} = this.props;
 
     const xIndex = i % uCount;
     const yIndex = (i - xIndex) / uCount;
 
-    return [(xIndex / (uCount - 1)) * 255, (yIndex / (vCount - 1)) * 255, 1];
+    return [(xIndex / (uCount - 1)) * 255, (yIndex / (vCount - 1)) * 255, 1] as $TODO;
   }
 
-  decodePickingColor([r, g, b]) {
+  decodePickingColor([r, g, b]: Color): number {
     if (b === 0) {
       return -1;
     }
-    return [r / 255, g / 255];
+    return [r / 255, g / 255] as $TODO;
```

Comment:
Probably not a huge deal, but TypeScript complains that these types defer from the base class.

Reply:
I would remove this method entirely, and rely on the `Layer` implementation. Then, modify `getPickingInfo` to reconstruct `[r, g, b]` from the `index`